backend/tests/integration/records/api/test_api_data.py

```diff
@@ -84,6 +84,36 @@ def test_data_update(inspire_app):
     assert control_number == record_updated_pid.pid_value
 
 
+def test_data_update_with_literature_record(inspire_app):
+    record_literature = create_record("lit")
+    record_literature_control_number = record_literature["control_number"]
+
+    data = faker.record("dat", with_control_number=True)
+    data["literature"] = [
+        {
+            "record": {
+                "$ref": f"http://localhost:8000/api/literature/{record_literature_control_number}"
+            }
+        }
+    ]
+    record = DataRecord.create(data)
+    data_update = {"dois": [{"value": "10.1000/UPDATED"}]}
+    data.update(data_update)
+    record.update(data)
+
+    control_number = str(record["control_number"])
+    record_updated_db = RecordMetadata.query.filter_by(id=record.id).one()
+
+    assert data == record_updated_db.json
+
+    record_updated_pid = PersistentIdentifier.query.filter_by(
+        pid_type="dat", pid_value=str(control_number)
```

Comment:
here you convert to string 2 times, you can just do control_number

Reply:
🐤 I copy-pasted the other test :P